### PR TITLE
boop: get_pending_transaction fn

### PR DESCRIPTION
**Pull request by antonbobkov**

### CODE REVIEW

File: src/pending_transactions.erl [hunk 1]
```diff
@@ -117,12 +117,50 @@ add_transaction(Transaction, PendingTx)
     end.
     
 
-%% @doc Get list of all the players in the container.
+%% doc Get list of all the players in the container.
 
-get_players(_) ->
-    ok.
+%% get_all_players(#pending_tx{player_map = PlMap}) ->
+%%     maps:keys(PlMap).
```
Comment (pdlla) [inline src/pending_transactions.erl]: just delete it?

File: src/pending_transactions.erl [hunk 2]
```diff
@@ -117,12 +117,50 @@ add_transaction(Transaction, PendingTx)
...
+		     R = maps:find(PlayerId, PlayersStartingNonceMap),
+		     case R of
+			 {ok, StartingNonce} -> ok;
+			 error -> StartingNonce = 0
+		     end,
+		     collect_transactions_from_player(TxMap, StartingNonce, TransactionListAcc)
+	     end,
+
+    % unsorted list of tuples {Counter, Transaction}
+    TransactionList = maps:fold(FoldFn, [], PlMap),
```
Comment (pdlla) [inline src/pending_transactions.erl]: cool!

File: src/pending_transactions.erl [hunk 3]
```diff
@@ -117,12 +117,50 @@ add_transaction(Transaction, PendingTx)
...
+
+
+    TupleCompareFn = fun({Counter1, _}, {Counter2, _}) -> Counter1 =< Counter2 end,
+    TupleMapFn = fun({_, Transaction}) -> Transaction end,
+
+    lists:map(TupleMapFn, lists:sort(TupleCompareFn, TransactionList)).
+
+%% extracts transactions from a player, in sequential order, starting at Nonce
+collect_transactions_from_player(TxMap, Nonce, TransactionListAcc) ->
```
Comment (pdlla) [inline src/pending_transactions.erl]: :o

Comment (pdlla) [reply src/pending_transactions.erl]: I would have some suggestions if this were haskell XD

File: test/pending_transactions_test.erl [hunk 4]
```diff
@@ -78,4 +83,7 @@ add_transaction_test() ->
     ?assertEqual([{0, 3}, {2, 4}, {3, 7}], get_nonce_counter_list(p1, PT1)),
     ?assertEqual([{0, 0}, {1, 5}, {2, 6}], get_nonce_counter_list(p2, PT1)),
 
+    ?assertEqual([{0, p2}, {0, p1}, {1, p2}, {2, p2}], get_all_transactions(#{}, PT1)),
+    ?assertEqual([{2, p1}, {1, p2}, {2, p2}, {3, p1}], get_all_transactions(#{p1 => 2, p2 => 1}, PT1)),
```
Comment (pdlla) [inline test/pending_transactions_test.erl]: nice!

File: src/pending_transactions.erl (hunk 1 above)
Comment (antonbobkov) [reply src/pending_transactions.erl]: best way to implement a function is to comment it out!

Comment (antonbobkov) [reply src/pending_transactions.erl]: such efficiency

File: src/pending_transactions.erl (hunk 3 above)
Comment (antonbobkov) [reply src/pending_transactions.erl]: that's actually pretty neat

Comment (pdlla) [reply src/pending_transactions.erl]: I came up with a better one 🙀 

unfoldr (\acc -> map (,acc+1) (lookup acc txmap)) nonce

(using TupleSections extension)

Comment (antonbobkov) [reply src/pending_transactions.erl]: unfold 😱

Comment (pdlla) [reply src/pending_transactions.erl]: incoming potato haskell pull request

Comment (pdlla) [reply src/pending_transactions.erl]: but it may be a while because I'm trying to implement unfoldr in erlang without recursion 😱

**Pull request merged**